### ISSUE TITLE
Unable to run flutter app on iPhone 12 simulator on my Mac M1

### ISSUE THREAD

**Issue by Asad4132** [flutter/flutter#76687]
Error:
```

Launching lib/main.dart on iPhone 12 Pro Max in debug mode...
Running pod install...
Running Xcode build...
Xcode build done.                                            8.7s
Failed to build iOS app
Error output from Xcode build:
↳
    objc[99436]: Class AMSupportURLConnectionDelegate is implemented in both ?? (0x1fd0f4188) and ?? (0x122af42b8). One of the two will be used. Which one is undefined.
    objc[99436]: Class AMSupportURLSession is implemented in both ?? (0x1fd0f41d8) and ?? (0x122af4308). One of the two will be used. Which one is undefined.
    ** BUILD FAILED **


Xcode's output:
↳
    /Users/asadurrehman/AndroidStudioProjects/IOS/teamsuite/ios/Runner/GeneratedPluginRegistrant.m:10:9: fatal error: module 'fluttertoast' not found
    @import fluttertoast;
     ~~~~~~~^~~~~~~~~~~~
    1 error generated.
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Constructing build description
    warning: Capabilities for Signing & Capabilities may not function correctly because its entitlements use a placeholder team ID. To resolve this, select a development team in the Runner editor. (in target 'Runner' from project 'Runner')

Could not build the application for the simulator.
Error launching application on iPhone 12 Pro Max.
```

my pubspec.yaml:
```

name: flutterapp
description: A new Flutter application.

# The following line prevents the package from being accidentally published to
# pub.dev using `pub publish`. This is preferred for private packages.
publish_to: 'none' # Remove this line if you wish to publish to pub.dev

# The following defines the version and build number for your application.
# A version number is three numbers separated by dots, like 1.2.43
# followed by an optional build number separated by a +.
# Both the version and the builder number may be overridden in flutter
# build by specifying --build-name and --build-number, respectively.
# In Android, build-name is used as versionName while build-number used as versionCode.
# Read more about Android versioning at https://developer.android.com/studio/publish/versioning
# In iOS, build-name is used as CFBundleShortVersionString while build-number used as CFBundleVersion.
# Read more about iOS versioning at
# https://developer.apple.com/library/archive/documentation/General/Reference/InfoPlistKeyReference/Articles/CoreFoundationKeys.html
version: 1.0.0+1

environment:
  sdk: ">=2.7.0 <3.0.0"

dependencies:
  flutter:
    sdk: flutter


  # The following adds the Cupertino Icons font to your application.
  # Use with the CupertinoIcons class for iOS style icons.
  cupertino_icons: ^1.0.0
  fluttertoast: ^7.1.8

dev_dependencies:
  flutter_test:
    sdk: flutter

# For information on the generic Dart part of this file, see the
# following page: https://dart.dev/tools/pub/pubspec

# The following section is specific to Flutter.
flutter:

  # The following line ensures that the Material Icons font is
  # included with your application, so that you can use the icons in
  # the material Icons class.
  uses-material-design: true

  # To add assets to your application, add an assets section, like this:
  # assets:
  #   - images/a_dot_burr.jpeg
  #   - images/a_dot_ham.jpeg

  # An image asset can refer to one or more resolution-specific "variants", see
  # https://flutter.dev/assets-and-images/#resolution-aware.

  # For details regarding adding assets from package dependencies, see
  # https://flutter.dev/assets-and-images/#from-packages

  # To add custom fonts to your application, add a fonts section here,
  # in this "flutter" section. Each entry in this list should have a
  # "family" key with the font family name, and a "fonts" key with a
  # list giving the asset and other descriptors for the font. For
  # example:
  # fonts:
  #   - family: Schyler
  #     fonts:
  #       - asset: fonts/Schyler-Regular.ttf
  #       - asset: fonts/Schyler-Italic.ttf
  #         style: italic
  #   - family: Trajan Pro
  #     fonts:
  #       - asset: fonts/TrajanPro.ttf
  #       - asset: fonts/TrajanPro_Bold.ttf
  #         weight: 700
  #
  # For details regarding fonts from package dependencies,
  # see https://flutter.dev/custom-fonts/#from-packages

```
**if I just remove the following line the app works:**

`  fluttertoast: ^7.1.8
`

similarly, I tried another library i.e. :

`  shared_preferences: ^0.5.12+4`

and I get a similar error i.e.

```
Xcode's output:
↳
    /Users/asadurrehman/AndroidStudioProjects/IOS/teamsuite/ios/Runner/GeneratedPluginRegistrant.m:10:9: fatal error: module 'shared_preferences' not found
    @import shared_preferences;
```

**Comment by TahaTesser** [comment]
Hi @Asad4132 
The first error is actually is a warning https://github.com/flutter/flutter/issues/72492#issuecomment-755760424

Can you please provide your `flutter doctor -v` and `flutter run --verbose` logs, and a [minimal complete reproducible code sample](https://stackoverflow.com/help/minimal-reproducible-example) with `shares_preferences`?
Thank you

**Comment by Asad4132** [comment]
flutter doctor -v:
```
[✓] Flutter (Channel stable, 1.22.6, on macOS 11.2 20D64 darwin-arm, locale en-PK)
    • Flutter version 1.22.6 at /Users/asadurrehman/Development/flutter
    • Framework revision 9b2d32b605 (5 weeks ago), 2021-01-22 14:36:39 -0800
    • Engine revision 2f0af37152
    • Dart version 2.10.5

 
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/asadurrehman/library/Android/sdk
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = /Users/asadurrehman/library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.4)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.4, Build version 12D4e
    • CocoaPods version 1.10.1

[!] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] Connected device (2 available)
    • SM G960F (mobile)          • 21411390430c7ece                     • android-arm64 • Android 10 (API 29)
    • iPhone 12 Pro Max (mobile) • 1B573028-07F6-47EB-8E4B-F6CE84226BA6 • ios           • com.apple.CoreSimulator.SimRuntime.iOS-14-4 (simulator)

! Doctor found issues in 1 category.

```

**Comment by Asad4132** [comment]
minimal complete reproducible code sample:

I just created a new flutter project and add the following dependency in pubspec.yaml:
`  shared_preferences: ^0.5.12+4`

I also tried with: 
`  fluttertoast: ^7.1.8
`

in both cases I am facing a similar type of error i.e.:

```
Xcode's output:
↳
    /Users/asadurrehman/AndroidStudioProjects/IOS/teamsuite/ios/Runner/GeneratedPluginRegistrant.m:10:9: fatal error: module 'fluttertoast' not found
    @import fluttertoast;
     ~~~~~~~^~~~~~~~~~~~
```

and in case of shared preferences:

```
Xcode's output:
↳
    /Users/asadurrehman/AndroidStudioProjects/IOS/teamsuite/ios/Runner/GeneratedPluginRegistrant.m:10:9: fatal error: module 'shared_preferences' not found
    @import shared_preferences;
```

**Comment by TahaTesser** [comment]
Hi @Asad4132 
I'm unable to reproduce this and @markusaksli-nc tried on M1 as well,  it might be a local issue  

Run these in the ios directory of your project
```bash
rm -rf Pods
rm -rf Podfile.lock
rm -rf ~/.pub-cache/hosted/pub.dartlang.org/
pod cache clean --all
flutter clean
flutter pub get
pod repo update
```

Thank you

**Comment by Asad4132** [comment]
@TahaTesser I just did that you told but still in vain. I have also updated flutter to new version but still getting the following error:
```
Xcode's output:
↳
    /Users/asadurrehman/AndroidStudioProjects/flutter_app/ios/Runner/GeneratedPluginRegistrant.m:10:9: fatal error: module 'shared_preferences' not found
    @import shared_preferences;
     ~~~~~~~^~~~~~~~~~~~~~~~~~
    1 error generated.
```

Flutter Doctor -v :
```
asadurrehman@192 flutter_app % flutter doctor -v
[✓] Flutter (Channel stable, 2.0.0, on macOS 11.2 20D64 darwin-arm, locale en-PK)
    • Flutter version 2.0.0 at /Users/asadurrehman/Development/flutter
    • Framework revision 60bd88df91 (2 days ago), 2021-03-03 09:13:17 -0800
    • Engine revision 40441def69
    • Dart version 2.12.0

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/asadurrehman/library/Android/sdk
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = /Users/asadurrehman/library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.4, Build version 12D4e
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] Connected device (2 available)
    • iPhone 12 Pro Max (mobile) • 1B573028-07F6-47EB-8E4B-F6CE84226BA6 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-4 (simulator)
    • Chrome (web)               • chrome                               • web-javascript • Google Chrome 88.0.4324.192

• No issues found!
```

**Comment by TahaTesser** [comment]
Hi @Asad4132 
We tried to replicate on Intel-mac as well as M1 Mac, using either package has no issue
Can you please try `flutters pub cache repair` and then `flutter clean`, run the app
Thank you

**Comment by Asad4132** [comment]
Hi @TahaTesser I have tried ` flutter pub cache repair` and then `flutter clean` but I am still getting the same error. Can we do something like you use my laptop remotely and check what's the real problem? The app runs perfectly on android devices but not for ios.

**Comment by TahaTesser** [comment]
@Asad4132 
That's not required, I will label this to be looked at by someone on the team
Thanks

**Comment by phhuan** [comment]
@TahaTesser  i had the same error "Module 'shared_preferences' not found" at this line of code "@import shared_preferences;". Do we have any solutions ? I updated shared_preferences version to "shared_preferences: ^2.0.5" and flutter is :
Flutter 2.0.4 • channel stable • https://github.com/flutter/flutter.git
Framework • revision b1395592de (2 weeks ago) • 2021-04-01 14:25:01 -0700
Engine • revision 2dce47073a
Tools • Dart 2.12.2

**Comment by TahaTesser** [comment]
@Asad4132 
Given your last message I feel safe to close this issue, if you disagree please write in the comments and I will reopen it. 
Thank you

Could everyone who still has this problem please file a new issue with the exact description of what happens, logs, and the output of `flutter doctor -v`.
All system setups can be slightly different, so it's always better to open new issues and reference related issues.
